src/gt4py/next/iterator/transforms/infer_domain.py

```diff
@@ -25,10 +25,29 @@
 from gt4py.next.utils import flatten_nested_tuple, tree_map
 
 
-DOMAIN: TypeAlias = domain_utils.SymbolicDomain | None | tuple["DOMAIN", ...]
+class DomainAccessDescriptor(eve.StrEnum):
```

Comment:
```suggestion
class _DomainAccessDescriptor(eve.StrEnum):
```
this is internal to the pass, correct?

Reply:
No, this can be used from the outside, it will for example appear in the `domain` annex attribute (used in `extract_tmp`) and could be used (I actually do this in the unused-return-from-scan fix I use in PMAP).